lib/manager/gradle/index.js

```diff
@@ -48,15 +55,8 @@ async function extractPackageFile(content, fileName, config) {
   const gradleFile = path.join(config.localDir, fileName);
   const baseDir = path.dirname(gradleFile);
 
-  if (isProjectRootGradle(fileName)) {
```

Comment:
This has nothing to do with the global variables, it's a potential bug.

This code needs to be executed always before extractDependenciesFromUpdatesReport() but if for whatever reason the projectRootGradle file is not the first one to process, the extractDependenciesFromUpdatesReport will be called first and it will not find any dependency. That's why I've moved to extractAllPackageFiles so now we are enforcing the order.

Reply:
Now that we call the function extractAllPackageFiles, we can refactor this code to process the root build.gradle first and avoid any race conditions like this.

Reply:
Yes, I moved the gradle execution in extractDependenciesFromUpdatesReport() so now is only executed once after all files has been extracted